chain/x/oracle/client/common/request_search.go

```diff
@@ -52,8 +52,15 @@ func QuerySearchLatestRequest(
 					}
 				}
 				if ok && rid != "" {
-					// TODO: Check if request has resolved.
-					return queryRequest(route, cliCtx, rid)
+					res, h, err := queryRequest(route, cliCtx, rid)
+					if err != nil {
```

Comment:
if err != nil we should just return error in my opinion. We never expect to get error.